client/core.py

```diff
@@ -0,0 +1,134 @@
+''' core functionality '''
+from block.block import Block, genesis_block
+from client.helpers import parse
+import cmd, sys, hashlib, json, os, codecs
+
+def initialize():
+    # initialize program on startup
+    # 1) check to see if conf appdata exists
+    # 2) if it does load it up then begin sync ( connect, get updates, etc )
+    # 3) if not, create new public key/wallet, files, mine genesis block, connect to some seed peers, sync
+    conf = load_conf()
+    if conf["height"] == 0:
+        # we need to save the genesis block
+        b = genesis_block()
+        # create a new folder dir for blocks
+        cwd = os.getcwd()
+        try:
+            os.makedirs(os.path.join(cwd, r'data'))
+        except OSError as e:
+            # folder exists or error
+            # print('error {0}'.format(e))
+            pass
+        save_block(b)
+        conf["height"] = 1
+    # print(json.dumps(conf, indent=4, sort_keys=True))
+    return conf
+
+def load_conf():
+    # loads the config file
+    try:
+        with open('abc.json') as file:
+            # read in the data
+            data = json.load(file)
+            file.close()
+            pass
+    except IOError as e:
+        # file does not exist or not able to read file
+        data = create_conf()
+        # print("not able to open conf file")
+    return data
+
+def create_conf():
+    # creates a new config

```

Comment:
We could just have this config file in our repository instead of creating it here.